Proofs/SHA512/Sha512_block_armv8.lean

```diff
@@ -102,6 +122,12 @@ theorem sha512_block_armv8_test_3_sym (s0 s_final : ArmState)
   simp_all only [state_simp_rules, -h_run]
   -- Symbolic simulation
   sym_n 4
+  case h_s1_sp_aligned =>
+    apply CheckSPAlignment_BitVecAdd_64_4
+    case x_aligned =>
+      exact h_s0_sp_aligned
+    case y_aligned =>
+      simp_all (config := {decide := true}) only [state_simp_rules, minimal_theory, bitvec_rules]
```

Comment:
```suggestion
  case h_s1_sp_aligned => apply Aligned_BitVecAdd_64_4 h_s0_sp_aligned (by decide)
```

- As seen previously, `CheckSPAlignment_BitVecAdd_64_4` is definitionally equal to `Aligned_BitVecAdd_64_4`, so we directly use `Aligned_BitVecAdd_64_4` if we like.
    - Since we made the other arguments implicit, we solve `x_aligned` with `h_s0_sp_aligned`. 
    - Since `y_aligned` is a concrete hypothesis about a concrete bitvector (`⊢ Aligned (18446744073709551600#64) 4`), we discharge it with `decide`. Hurray on us for proving that `Aligned` is decidable :)

Reply:
Excellent!  As I mentioned in a comment to Shilpi on this PR, I'd like to understand why we need this case at all.  We should be able to automate application of alignment lemmas.